Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.CssEditor/scripts/CssEditor.js

```diff
@@ -97,26 +98,82 @@ define([
             var panelHeight = $('#CssEditor-panel').height();
             if (panelHeight > 400) {
-                cssEditor.setSize("100%", panelHeight - 330);
+                $('#monaco-editor').height(panelHeight - 400);
             }
 
             if (typeof callback === 'function') {
                 callback();
             }
 
-            cssEditor.on("blur", function(cm) {
+            $('#monaco-editor').on("blur", function(cm) {
                 cm.save();
                 return true;
             });
 
             viewModel.portal.subscribe(portalChanged);
         };
 
+        var initCssEditor = function() {
+            var monacoEditorLoaderScript = document.createElement('script');
+            monacoEditorLoaderScript.type = 'text/javascript';
+            monacoEditorLoaderScript.src = '/Resources/Shared/components/MonacoEditor/loader.js';
+            document.body.appendChild(monacoEditorLoaderScript);
+
+            require.config({ paths: { 'vs': '/Resources/Shared/components/MonacoEditor' }});
+            require(['vs/editor/editor.main'], function(monaco) {
+        
+                self.MonacoEnvironment = {
+                    getWorkerUrl: function (moduleId, label) {
+                        if (label === 'typescript' || label === 'javascript') {
+                        return `data:text/javascript;charset=utf-8,${encodeURIComponent(`
+                            importScripts('${process.env.ASSET_PATH}/typescript.worker.js');`
+                        )}`;
+                        }
+        
+                    return `data:text/javascript;charset=utf-8,${encodeURIComponent(`
+                        importScripts('${process.env.ASSET_PATH}/editor.worker.js');`
+                    )}`;
+                    }
+                }
+
+                cssEditor = monaco.editor;
+                cssContent = cssEditor.createModel("", "css");
+                initMonacoEditor();
+                getStyleSheet();
+            });
+
+        }
+
+        var initMonacoEditor = function() {
+            var theme = "vs-light";
+            if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+                theme = "vs-dark";
+            }
+            window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
+                theme = e.matches ? "vs-dark" : "vs-light";
+            });
+
+            var monacoEditor = cssEditor.create(document.getElementById("monaco-editor"), {
+                model: cssContent,
+                language: "css",
+                wordWrap: 'wordWrapColumn',
+                wordWrapColumn: 80,
+                wordWrapMinified: true,
+                wrappingIndent: "indent",
+                lineNumbers: "on",
+                roundedSelection: false,
+                scrollBeyondLastLine: false,
+                readOnly: false,
+                theme: theme,
+                automaticLayout: true
+            });
```

Comment:
See comment about about ReactJS component.

Reply:
See #4132.  This is a stepping stone solution.  Moving this to a proper module with package management will a future task.